packages/core/src/scripts/utils/read-configuration.ts

```diff
@@ -0,0 +1,65 @@
+import { Site } from "@frontity/file-settings/types";
+import { pathExists } from "fs-extra";
+import { uniq } from "lodash";
+
+/**
+ * Creates string that represents a package path.
+ *
+ * @param name - The package name.
+ * @param file - The file name.
+ * @returns The path.
+ */
+const createPath = (name, file) =>
+  `${process.env.CWD}/node_modules/${name}/${file}`;
+
+/**
+ * Checks if a given package has a configuration file.
+ *
+ * @param name - The package name.
+ * @returns If a package has a configuration file.
+ */
+const packageHasConfig = async (name) => {
+  return (
+    (await pathExists(createPath(name, "frontity.config.ts"))) ||
+    (await pathExists(createPath(name, "frontity.config.js")))
+  );
+};
+
+/**
+ * This function applies the configuration from packages.
+ *
+ * @param sites - The sites.
+ * @returns A dictionary split by the exported configuration function.
+ */
+export const readConfigurationsFromConfigFiles = async (
+  sites: Site[]
+): Promise<Record<string, []>> => {
+  const packages = uniq(
+    sites.reduce((out, site) => {
+      return out.concat(site.packages);
+    }, [])
+  );
+  const dictionary = {};
+
+  for (const i in packages) {
+    const name = packages[i];
```

Comment:
Totally out of curiosity: is it better to use `for..in` loop instead of the `for...of`?

```suggestion
  for (const name of packages) {
```

Reply:
I mean, _better_ is subjective 😄  the thing with `for..of` is that is getting transpiled as `for..in` is not. For example:

```js
// Does not get transpiled. What you write is what you get.
for (const i in list) {
  const entry = list[i];
}

// This one instead gets transpiled
for (const entry of list) {
}

// Results in:
for (var _i = 0, _list = list; _i < _list.length; _i++) {
  var _entry = _list[_i];
}
```
In the end, you still get the `get` property accessor called, so, not really sure if is it better to use `for..in`? 🤷  I really have no preference. `for..in` was shorter to type.

Let me know what you think.

Reply:
I was just curious if it had any performance/size implications 🙂

Thanks for your answer.